lib/web_ui/test/engine/platform_dispatcher/platform_dispatcher_test.dart

```diff
@@ -365,6 +365,22 @@ void testMain() {
       expect(onMetricsChangedCalled, isFalse);
       expect(view1.isDisposed, isTrue);
     });
+
+    test('invokeOnViewFocusChange calls onViewFocusChange', () {
+      final EnginePlatformDispatcher dispatcher = EnginePlatformDispatcher();
+      final List<ui.ViewFocusEvent> dispatchedViewFocusEvents = <ui.ViewFocusEvent>[];
+      const ui.ViewFocusEvent viewFocusEvent = ui.ViewFocusEvent(
+        viewId: 0,
+        focusState: ui.ViewFocusState.focused,
+        focusDirection: ui.ViewFocusDirection.undefined,
+      );
+
+      dispatcher.onViewFocusChange = dispatchedViewFocusEvents.add;
+      dispatcher.invokeOnViewFocusChange(viewFocusEvent);
+
+      expect(dispatchedViewFocusEvents, hasLength(1));
+      expect(dispatchedViewFocusEvents.single, viewFocusEvent);
```

Comment:
Bonus points for testing that the zone is preserved, as promised by the API docs.

Reply:
Bonus points claimed.

Reply:
Great call, done!